specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimapis.json

```diff
@@ -1399,15 +1660,174 @@
+          "description": "Indicates if API revision is accessible via the gateway.",
+          "readOnly": true
         }
       },
       "description": "API base contract details."
     },
+    "ApiRevisionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiRevisionContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Api Revision list representation."
+    },
+    "ApiRevisionContract": {
+      "properties": {
+        "apiId": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Identifier of the API Revision."
+        },
+        "apiRevision": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Revision number of API."
+        },
+        "createdDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision was created."
+        },
+        "updatedDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision were updated."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the API Revision."
+        },
+        "privateUrl": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gateway URL for accessing the non-current API Revision."
+        },
+        "isOnline": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "Indicates if API revision is the current api revision."
+        },
+        "isCurrent": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "Indicates if API revision is accessible via the gateway."
+        }
+      }
+    },
+    "ApiRevisionInfoContract" : {
+        "properties": {
+          "sourceApiId": {
+            "type": "string",
+            "description": "Id of API to be used to create the revision from."
+          },
+          "ApiVersionName": {
+            "type": "string",
+            "description": "Version identifier for the new API Version."
+          },
+          "ApiVersionDescription": {
+            "type": "string",
+            "description": "Description of new API Version"
+          },
+          "ApiRevisionDescription": {
+            "type": "string",
+            "description": "Description of new API Revision."
+          }
+        }
+    },
+    "ApiVersionSetContract" : {
+      "type": "object",
+      "description": "An API Version Set contains the common configuration for a set of API Versions relating ",
+      "properties": {
+        "Id" : {
```

Comment:
>Id [](start = 9, length = 2)

not sure if Id comes out, when we call via ARM. example would help here.

Reply:
I will run some tests to try it.  The Id better appear because otherwise we will create a new versionset on every update.